Add windows PCI filter driver doc (#2048)
chipsec/chipsec: docs/sphinx/installation/InstallWindows.rst

   ``sc stop chipsec``
   ``sc delete chipsec``

Windows PCI Filter Driver
-------------------------

**Filter driver background**

> Since July 31, Microsoft has released Windows 2020-KB4568831 (OS Build 19041.423) Preview. you should not access the PCI configuration space using the legacy API, it might result in the Windows BSOD (Blue Screen of Death). The BSOD trigger condition is “Windows version >= (OS Build 19041.423) && Secure Devices (SDEV) ACPI table && VBS enabled”. Therefore, we have introduced a PCI filter driver that collaborates with the chipsec main driver to access the PCI configuration space. If you do not meet these conditions, you do not need to install the PCI filter driver, and you can use the chipsec main helper driver directly.

Reviewer: ```suggestion
Since July 31, 2020 Microsoft has released Windows 2020-KB4568831 (OS Build 19041.423) Preview. Microsoft recommends to not access the PCI configuration space using the legacy API, as it might result in the Windows BSOD (Blue Screen of Death). The BSOD trigger condition is “Windows version >= (OS Build 19041.423) && Secure Devices (SDEV) ACPI table && VBS enabled”. Therefore, CHIPSEC now includes a PCI filter driver which supplements the original CHIPSEC Windows Driver to access the PCI configuration space. A system only requires the PCI Filter Driver if the conditions above are met. 
```

Author: done

> Windows devices that receive the July 31, 2020-KB4568831 (OS Build 19041.423) Preview or later updates restrict how processes 

Reviewer: ```suggestion
Windows devices that receive the 2020-KB4568831 (OS Build 19041.423) Preview or later updates restrict how processes 
```

Author: done

Credential Guard. The new restrictions are designed to prevent malicious processes from modifying the configuration space of 
secure devices. Device drivers or other system processes must not try to manipulate the configuration space of any PCI devices, 
except by using the Microsoft-provided bus interfaces or IRP. If a process tries to access PCI configuration space in an unsupported 
manner (such as by parsing MCFG table and mapping configuration space to virtual memory), Windows denies access to the process and 
generates a Stop error. For more detail please refer below link: https://learn.microsoft.com/en-us/troubleshoot/windows-client/performance/stop-error-lenovo-thinkpad-kb4568831-uefi

**Filter Driver and Main Helper Driver Architecture**

.. figure:: ../_images/driver_architecture.png
   :alt: CHIPSEC Main & Filter Drvier Architecture
   :width: 1100
   :height: 1000

   CHIPSEC Main & Filter Drvier Architecture

Install PCI Filter Driver
-------------------------
**Check The Filter Drvier Files**

.. figure:: ../_images/driver_files.png
   :alt: Check The Filter Drvier Files
   :width: 1000
   :height: 470

>    Check The Filter Drvier Files

Reviewer: ```suggestion
   Locate the Filter Driver Files: chipsec/helper/windows/windows_amd64/
```

Author: done

> **Update The PCI Device Driver**

Reviewer: ```suggestion
**Update The PCI Device Driver from Device Manager**
```

Author: done

.. figure:: ../_images/update_driver.png
   :alt: Update The PCI Device Driver
   :width: 950
   :height: 760

   Update The PCI Device Driver

**Browse The PCI Filter Driver**

 .. figure:: ../_images/browse_driver.png
   :alt: Browse The PCI Filter Driver
   :width: 950
   :height: 720

   Browse The PCI Filter Driver

> **Pickup The PCI Filter Driver**

Reviewer: ```suggestion
**Manually Select the PCI Filter Driver**
```

Author: done
